webpack/webpack5-localization-plugin/src/test/LocalizedAsyncDynamicFormatWithNoLocaleFallback.test.ts

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+jest.disableAutomock();
+import { resolve } from 'path';
+import { promisify } from 'util';
+
+import webpack, { type Compiler, type Stats } from 'webpack';
+import { Volume } from 'memfs/lib/volume';
+
+import { LocalizationPlugin } from '../LocalizationPlugin';
+import type { ILocalizationPluginOptions, ILocalizationStats } from '../interfaces';
+import { MemFSPlugin } from './MemFSPlugin';
+
+async function testLocalizedAsyncDynamicInner(minimize: boolean): Promise<void> {
+  const memoryFileSystem: Volume = new Volume();
+  memoryFileSystem.fromJSON(
+    {
+      '/a/package.json': '{ "name": "a", "sideEffects": ["entry.js", "async.js"] }',
+      '/a/async.js': `import strings1 from './strings1.resjson'; import strings2 from './strings2.resjson'; console.log(strings1.test, strings2.another);`,
+      '/a/entry.js': `import(/* webpackChunkName: 'async' */ './async');`,
+      '/a/strings1.resjson': `{"test":"blah","_test.comment":"A string"}`,
+      '/a/strings2.resjson': `{"another":"something else","_another.comment":"Another string"}`,
+      '/b/entry.js': `console.log('hello world');`
+    },
+    '/'
+  );
+
+  let localizationStats: ILocalizationStats | undefined;
+  function statsCallback(stats: ILocalizationStats): void {
+    localizationStats = stats;
+  }
+
+  const resJsonLoader: string = resolve(__dirname, '../loaders/resjson-loader.js');
+  const options: ILocalizationPluginOptions = {
+    localizedData: {
+      defaultLocale: {
+        localeName: 'en-us'
+      },
+      translatedStrings: {
+        foo: {
+          '/a/strings1.resjson': {
+            test: 'baz'
+          },
+          '/a/strings2.resjson': {
+            another: 'some random translation'
+          }
+        }
+      }
+    },
+    runtimeLocaleExpression: 'self.__locale',
+    localizationStats: {
+      callback: statsCallback
+    },
+    formatLocaleForFilename: (locale: string) => {
+      if (locale === 'none') {
+        return '';
+      } else {
+        return locale;
+      }
+    }
```

Comment:
The initial expectation of how to use this was to have `output.filename` be `[locale][name].js` and have this function return `''` for `'none'` and `${locale}/` for all other values. This would then entail `runtimeLocaleExpression` being `self.__locale + "/"`.

Reply:
So does this mean that `runtimeLocaleExpression` must always be set when `formatLocaleForFilename` is also set?